### PR TITLE
JDK-8300242 : Replace NULL with nullptr in share/code/

### PR DESCRIPTION
**Pull request by jdksjolen**
Do the conversion in the share/code/ sub-directory and all of its files.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8300242](https://bugs.openjdk.org/browse/JDK-8300242): Replace NULL with nullptr in share/code/

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to [252dcd15](https://git.openjdk.org/jdk/pull/12031/files/252dcd15cba96bd4c5c54b70abf66bbd61762439)
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**) ⚠️ Review applies to [252dcd15](https://git.openjdk.org/jdk/pull/12031/files/252dcd15cba96bd4c5c54b70abf66bbd61762439)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12031/head:pull/12031` \
`$ git checkout pull/12031`

Update a local copy of the PR: \
`$ git checkout pull/12031` \
`$ git pull https://git.openjdk.org/jdk pull/12031/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12031`

View PR using the GUI difftool: \
`$ git pr show -t 12031`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12031.diff">https://git.openjdk.org/jdk/pull/12031.diff</a>

</details>

### CODE REVIEW

**Review by vnkozlov — APPROVED** [review]
Looks good.

**Comment by vnkozlov** [comment]
@jdksjolen  I noticed that your repo is not configured for GHA testing. Please, enable it.

File: src/hotspot/share/code/nmethod.cpp [hunk 1]
```diff
@@ -92,7 +92,7 @@
 #define DTRACE_METHOD_UNLOAD_PROBE(method)                                \
   {                                                                       \
     Method* m = (method);                                                 \
-    if (m != NULL) {                                                      \
+    if (m != nullptr) {                                                      \
```
Comment (TobiHartmann) [inline src/hotspot/share/code/nmethod.cpp:95]:
```suggestion
    if (m != nullptr) {                                                   \
```

**Review by TobiHartmann — APPROVED** [review]
Otherwise looks good to me.

**Comment by jdksjolen** [comment]
Thank you for the reviews, I will integrate when testing passes.

**Comment by jdksjolen** [comment]
/integrate

Tests passed, integrating.

**Pull request closed without merging**